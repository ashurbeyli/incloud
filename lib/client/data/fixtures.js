export const dumbTask = {
  description: '',
  pieces: [],
};

export const task = {
  createdAt: '2018-04-19T22:07:34.639Z',
  description: 'asdfafa',
  pieces: [],
  updatedAt: '2018-04-19T22:07:34.640Z',
  user_id: '5ad65aa44212e29c4a922af7',
  __v: 0,
  _id: '5ad91326115be0385d527a1b'
};

export const tasks = [
  task,
  {
    createdAt: '2018-04-19T22:07:34.639Z',
    description: 'asdfafa',
    pieces: [],
    updatedAt: '2018-04-19T22:07:34.640Z',
    user_id: '5ad65aa44212e29c4a922af7',
    __v: 0,
    _id: '5ad916889a0c1f39b919eff7'
  },
];