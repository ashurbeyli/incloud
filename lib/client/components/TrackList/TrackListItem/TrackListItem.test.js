import React from 'react';
import Moment from 'react-moment';

import { TrackListItem } from './';
import { task } from '../../../data/fixtures';

const props = {
  task,
  onDelete: (id) => {}
};

describe('TrackListItem', () => {
  const wrapper = shallow(<TrackListItem {...props}/>);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should have task with description of "${props.task.description}"`, () => {
    expect(wrapper.find('Col.tracklist-item__description').render().text()).toEqual(props.task.description);
  });

});