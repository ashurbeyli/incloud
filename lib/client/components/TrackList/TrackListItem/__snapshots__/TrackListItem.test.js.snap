// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListItem should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onDelete={[Function]}
    task={
      Object {
        "__v": 0,
        "_id": "5ad91326115be0385d527a1b",
        "createdAt": "2018-04-19T22:07:34.639Z",
        "description": "asdfafa",
        "pieces": Array [],
        "updatedAt": "2018-04-19T22:07:34.640Z",
        "user_id": "5ad65aa44212e29c4a922af7",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "5ad91326115be0385d527a1b",
    "nodeType": "class",
    "props": Object {
      "bsClass": "list-group-item",
      "children": <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <Row
          bsClass="row"
          className="show-grid"
          componentClass="div"
        >
          <Col
            bsClass="col"
            className="tracklist-item__id"
            componentClass="div"
            md={1}
            xs={12}
          >
            5ad91326115be0385d527a1b
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__description"
            componentClass="div"
            md={5}
            xs={12}
          >
            asdfafa
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__datetime"
            componentClass="div"
            md={2}
            xs={12}
          >
            <t
              ago={false}
              calendar={false}
              decimal={false}
              element="time"
              filter={[Function]}
              format="DD MMM YYYY hh:mm"
              fromNow={false}
              interval={60000}
              onChange={[Function]}
              toNow={false}
              unit={null}
              unix={false}
              utc={false}
            >
              2018-04-19T22:07:34.639Z
            </t>
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__datetime"
            componentClass="div"
            md={1}
            xs={12}
          >
            00:00:00
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__actions"
            componentClass="div"
            md={2}
            xs={6}
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="small"
              bsStyle="danger"
              disabled={false}
              onClick={[Function]}
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="trash"
              />
            </Button>
          </Col>
        </Row>
      </Grid>,
      "className": "tracklist-item",
      "listItem": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          className="show-grid"
          componentClass="div"
        >
          <Col
            bsClass="col"
            className="tracklist-item__id"
            componentClass="div"
            md={1}
            xs={12}
          >
            5ad91326115be0385d527a1b
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__description"
            componentClass="div"
            md={5}
            xs={12}
          >
            asdfafa
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__datetime"
            componentClass="div"
            md={2}
            xs={12}
          >
            <t
              ago={false}
              calendar={false}
              decimal={false}
              element="time"
              filter={[Function]}
              format="DD MMM YYYY hh:mm"
              fromNow={false}
              interval={60000}
              onChange={[Function]}
              toNow={false}
              unit={null}
              unix={false}
              utc={false}
            >
              2018-04-19T22:07:34.639Z
            </t>
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__datetime"
            componentClass="div"
            md={1}
            xs={12}
          >
            00:00:00
          </Col>
          <Col
            bsClass="col"
            className="tracklist-item__actions"
            componentClass="div"
            md={2}
            xs={6}
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="small"
              bsStyle="danger"
              disabled={false}
              onClick={[Function]}
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="trash"
              />
            </Button>
          </Col>
        </Row>,
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": Array [
            <Col
              bsClass="col"
              className="tracklist-item__id"
              componentClass="div"
              md={1}
              xs={12}
            >
              5ad91326115be0385d527a1b
            </Col>,
            <Col
              bsClass="col"
              className="tracklist-item__description"
              componentClass="div"
              md={5}
              xs={12}
            >
              asdfafa
            </Col>,
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={2}
              xs={12}
            >
              <t
                ago={false}
                calendar={false}
                decimal={false}
                element="time"
                filter={[Function]}
                format="DD MMM YYYY hh:mm"
                fromNow={false}
                interval={60000}
                onChange={[Function]}
                toNow={false}
                unit={null}
                unix={false}
                utc={false}
              >
                2018-04-19T22:07:34.639Z
              </t>
            </Col>,
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={1}
              xs={12}
            >
              00:00:00
            </Col>,
            <Col
              bsClass="col"
              className="tracklist-item__actions"
              componentClass="div"
              md={2}
              xs={6}
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="small"
                bsStyle="danger"
                disabled={false}
                onClick={[Function]}
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="trash"
                />
              </Button>
            </Col>,
          ],
          "className": "show-grid",
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": "5ad91326115be0385d527a1b",
              "className": "tracklist-item__id",
              "componentClass": "div",
              "md": 1,
              "xs": 12,
            },
            "ref": null,
            "rendered": "5ad91326115be0385d527a1b",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": "asdfafa",
              "className": "tracklist-item__description",
              "componentClass": "div",
              "md": 5,
              "xs": 12,
            },
            "ref": null,
            "rendered": "asdfafa",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <t
                ago={false}
                calendar={false}
                decimal={false}
                element="time"
                filter={[Function]}
                format="DD MMM YYYY hh:mm"
                fromNow={false}
                interval={60000}
                onChange={[Function]}
                toNow={false}
                unit={null}
                unix={false}
                utc={false}
              >
                2018-04-19T22:07:34.639Z
              </t>,
              "className": "tracklist-item__datetime",
              "componentClass": "div",
              "md": 2,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "ago": false,
                "calendar": false,
                "children": "2018-04-19T22:07:34.639Z",
                "decimal": false,
                "element": "time",
                "filter": [Function],
                "format": "DD MMM YYYY hh:mm",
                "fromNow": false,
                "interval": 60000,
                "onChange": [Function],
                "toNow": false,
                "unit": null,
                "unix": false,
                "utc": false,
              },
              "ref": null,
              "rendered": "2018-04-19T22:07:34.639Z",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": "00:00:00",
              "className": "tracklist-item__datetime",
              "componentClass": "div",
              "md": 1,
              "xs": 12,
            },
            "ref": null,
            "rendered": "00:00:00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="small"
                bsStyle="danger"
                disabled={false}
                onClick={[Function]}
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="trash"
                />
              </Button>,
              "className": "tracklist-item__actions",
              "componentClass": "div",
              "md": 2,
              "xs": 6,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsSize": "small",
                "bsStyle": "danger",
                "children": <Glyphicon
                  bsClass="glyphicon"
                  glyph="trash"
                />,
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "glyphicon",
                  "glyph": "trash",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "5ad91326115be0385d527a1b",
      "nodeType": "class",
      "props": Object {
        "bsClass": "list-group-item",
        "children": <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <Row
            bsClass="row"
            className="show-grid"
            componentClass="div"
          >
            <Col
              bsClass="col"
              className="tracklist-item__id"
              componentClass="div"
              md={1}
              xs={12}
            >
              5ad91326115be0385d527a1b
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__description"
              componentClass="div"
              md={5}
              xs={12}
            >
              asdfafa
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={2}
              xs={12}
            >
              <t
                ago={false}
                calendar={false}
                decimal={false}
                element="time"
                filter={[Function]}
                format="DD MMM YYYY hh:mm"
                fromNow={false}
                interval={60000}
                onChange={[Function]}
                toNow={false}
                unit={null}
                unix={false}
                utc={false}
              >
                2018-04-19T22:07:34.639Z
              </t>
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={1}
              xs={12}
            >
              00:00:00
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__actions"
              componentClass="div"
              md={2}
              xs={6}
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="small"
                bsStyle="danger"
                disabled={false}
                onClick={[Function]}
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="trash"
                />
              </Button>
            </Col>
          </Row>
        </Grid>,
        "className": "tracklist-item",
        "listItem": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": <Row
            bsClass="row"
            className="show-grid"
            componentClass="div"
          >
            <Col
              bsClass="col"
              className="tracklist-item__id"
              componentClass="div"
              md={1}
              xs={12}
            >
              5ad91326115be0385d527a1b
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__description"
              componentClass="div"
              md={5}
              xs={12}
            >
              asdfafa
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={2}
              xs={12}
            >
              <t
                ago={false}
                calendar={false}
                decimal={false}
                element="time"
                filter={[Function]}
                format="DD MMM YYYY hh:mm"
                fromNow={false}
                interval={60000}
                onChange={[Function]}
                toNow={false}
                unit={null}
                unix={false}
                utc={false}
              >
                2018-04-19T22:07:34.639Z
              </t>
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__datetime"
              componentClass="div"
              md={1}
              xs={12}
            >
              00:00:00
            </Col>
            <Col
              bsClass="col"
              className="tracklist-item__actions"
              componentClass="div"
              md={2}
              xs={6}
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsSize="small"
                bsStyle="danger"
                disabled={false}
                onClick={[Function]}
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="trash"
                />
              </Button>
            </Col>
          </Row>,
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": Array [
              <Col
                bsClass="col"
                className="tracklist-item__id"
                componentClass="div"
                md={1}
                xs={12}
              >
                5ad91326115be0385d527a1b
              </Col>,
              <Col
                bsClass="col"
                className="tracklist-item__description"
                componentClass="div"
                md={5}
                xs={12}
              >
                asdfafa
              </Col>,
              <Col
                bsClass="col"
                className="tracklist-item__datetime"
                componentClass="div"
                md={2}
                xs={12}
              >
                <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD MMM YYYY hh:mm"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
                >
                  2018-04-19T22:07:34.639Z
                </t>
              </Col>,
              <Col
                bsClass="col"
                className="tracklist-item__datetime"
                componentClass="div"
                md={1}
                xs={12}
              >
                00:00:00
              </Col>,
              <Col
                bsClass="col"
                className="tracklist-item__actions"
                componentClass="div"
                md={2}
                xs={6}
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="small"
                  bsStyle="danger"
                  disabled={false}
                  onClick={[Function]}
                >
                  <Glyphicon
                    bsClass="glyphicon"
                    glyph="trash"
                  />
                </Button>
              </Col>,
            ],
            "className": "show-grid",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": "5ad91326115be0385d527a1b",
                "className": "tracklist-item__id",
                "componentClass": "div",
                "md": 1,
                "xs": 12,
              },
              "ref": null,
              "rendered": "5ad91326115be0385d527a1b",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": "asdfafa",
                "className": "tracklist-item__description",
                "componentClass": "div",
                "md": 5,
                "xs": 12,
              },
              "ref": null,
              "rendered": "asdfafa",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <t
                  ago={false}
                  calendar={false}
                  decimal={false}
                  element="time"
                  filter={[Function]}
                  format="DD MMM YYYY hh:mm"
                  fromNow={false}
                  interval={60000}
                  onChange={[Function]}
                  toNow={false}
                  unit={null}
                  unix={false}
                  utc={false}
                >
                  2018-04-19T22:07:34.639Z
                </t>,
                "className": "tracklist-item__datetime",
                "componentClass": "div",
                "md": 2,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "ago": false,
                  "calendar": false,
                  "children": "2018-04-19T22:07:34.639Z",
                  "decimal": false,
                  "element": "time",
                  "filter": [Function],
                  "format": "DD MMM YYYY hh:mm",
                  "fromNow": false,
                  "interval": 60000,
                  "onChange": [Function],
                  "toNow": false,
                  "unit": null,
                  "unix": false,
                  "utc": false,
                },
                "ref": null,
                "rendered": "2018-04-19T22:07:34.639Z",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": "00:00:00",
                "className": "tracklist-item__datetime",
                "componentClass": "div",
                "md": 1,
                "xs": 12,
              },
              "ref": null,
              "rendered": "00:00:00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsSize="small"
                  bsStyle="danger"
                  disabled={false}
                  onClick={[Function]}
                >
                  <Glyphicon
                    bsClass="glyphicon"
                    glyph="trash"
                  />
                </Button>,
                "className": "tracklist-item__actions",
                "componentClass": "div",
                "md": 2,
                "xs": 6,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsSize": "small",
                  "bsStyle": "danger",
                  "children": <Glyphicon
                    bsClass="glyphicon"
                    glyph="trash"
                  />,
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "glyphicon",
                    "glyph": "trash",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
